### DISCUSSION TITLE
Is Red Hat Telemetry a required depedency for all adopters of LSP4iJ?

### DISCUSSION THREAD

**Discussion by mrglavas** [redhat-developer/lsp4ij#287] [Q&A] [answered]
We have been working on a proof-of-concept [1] to use the LSP4iJ plug-in from the Marketplace and while discussing progress with our developers they mentioned that they had to accept a dependency on Red Hat Telemetry to get this working and that this generates prompts to the user to install it. I've read the documentation and understand that users can opt-out but wonder if this dependency could be made optional for plug-ins that adopt LSP4iJ. Given that we're currently embedding a fork of LSP4iJ in our plug-in releases, we're not currently collecting usage data. We're not sure we want to start doing that or even have that as an option.

[1] https://github.com/OpenLiberty/liberty-tools-intellij/tree/useLSP4IJmarket

**Comment by fbricon** [comment]
First of all, Red Hat Telemetry is opt-in. We'll need to discuss this internally at Red Hat and will get back to you on that.

**Comment by yeekangc** [comment]
Thank you, Fred.  Let us know.  As a common component, we do think RH Telemetry should at least be made optional since not all adopters will want to enable it, even if it is an opt-in option for the end users.

We do look forward to adopt lsp4ij as soon as possible.  Then, it can be a blocker.

**Comment by fbricon** [comment, marked as answer]
So we decided to go forward and make telemetry optional, see https://github.com/redhat-developer/lsp4ij/pull/292

**Comment by mrglavas** [comment]
@fbricon Thank you for making the Telemetry dependency optional. Our team has verified and tested the change with a recent nightly build.
